### PR TITLE
Thread application_properties into Message within eventData

### PR DESCRIPTION
**Pull request by KieranBrantnerMagee**
This fixes a group of related issues, including [this bug.](https://github.com/Azure/azure-sdk-for-python/issues/8401)

Ensures that application_properties is properly transmitted, providing the encode_data overload as well.

Update UAMQP library version

Ensure proper conversion of timeout into milliseconds for updated UAMQP version

Add a unit test to validate this test failure, and reenable the test_receive_batch_with_app_prop_sync test

Add history and increment version ID

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/eventhub/azure-eventhubs/azure/eventhub/common.py [hunk 1]
```diff
@@ -258,6 +258,9 @@ def body_as_json(self, encoding='UTF-8'):
         except Exception as e:
             raise TypeError("Event data is not compatible with JSON type: {}".format(e))
 
+    def encode_message(self):
+        return self.message.encode_message()
+
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhubs/azure/eventhub/common.py]: Do we want to make this public? What's the scenario requiring this?

File: sdk/eventhub/azure-eventhubs/setup.py [hunk 2]
```diff
@@ -62,7 +62,7 @@
         "tests",
         "tests.asynctests"]),
     install_requires=[
-        'uamqp~=1.1.0',
+        'uamqp~=1.2.3',
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhubs/setup.py]: Does this version include the socket timeout fix that was causing the "~30 min send blackout" issue?

File: sdk/eventhub/azure-eventhubs/azure/eventhub/common.py (hunk 1 above)
Comment (KieranBrantnerMagee) [reply sdk/eventhub/azure-eventhubs/azure/eventhub/common.py]: This is apparently the "plugin" required by the underlying UAMQP bits that we brought forward to support this scenario. (was stolen wholesale from Adam's track2 PR that fixes this bug)

File: sdk/eventhub/azure-eventhubs/setup.py (hunk 2 above)
Comment (KieranBrantnerMagee) [reply sdk/eventhub/azure-eventhubs/setup.py]: I believe so, if I'm reading the latest in the email thread properly.  Uamqp 1.2.3 + the seconds->milliseconds conversion in EH proper should address the sending timeout failure.

@yunhaoling feel free to shout if I misinterpreted that thread or failures.

Comment (yunhaoling) [reply sdk/eventhub/azure-eventhubs/setup.py]: Just adding more context here: It's the receive blackout and simply introducing idle_timeout would bring new issues.
Need more efforts into the blackout issue.

**Pull request merged**